lib/symboldatabase.cpp

```diff
@@ -2102,7 +2102,10 @@ Function::Function(const Tokenizer *mTokenizer,
     }
 
     // class constructor/destructor
-    else if (tokenDef->str() == scope->className && scope->type != Scope::ScopeType::eNamespace) {
+    else if (((tokenDef->str() == scope->className) ||
+              (tokenDef->str().substr(0, scope->className.size()) == scope->className &&
+               tokenDef->str()[scope->className.size() + 1] == '<')) &&
```

Comment:
Don't we need to check `tokenDef->str().size() > scope->className.size() + 1`?

Reply:
Yes.